imports/plugins/included/shipping-shippo/server/methods/shippo.js

```diff
@@ -349,6 +372,59 @@ export const methods = {
 
     return updatingResult;
   },
+  /**
+   * Formats and sends a cart address to Shippo to for validation
+   * @param {String} address - The address to validate
+   * @return {Object} validateAddress - The validation result from Shippo
+   */
+  "shippo/addressValidation"(address) {
+    check(address, Object);
+
+    const apiKey = getApiKey(Reaction.getShopId());
+
+    if (!apiKey) {
+      const msg = "No Shippo API key was found.";
+      Logger.fatal(msg);
+      throw new Meteor.Error("shippo-api-error", msg);
+    }
+
+    // get the buyer's email
+    const buyer = Accounts.findOne({ _id: this.userId }, { fields: { emails: 1 } });
+
+    if (!address) {
+      const msg = "The 'shipping' property of this cart is either missing or incomplete.";
+      Logger.error(msg);
+      throw new Meteor.Error("address-validation-error", msg);
+    }
+
+    const shop = Shops.findOne({ _id: Reaction.getShopId() }, { fields: { emails: 1 } });
+
+    // TODO take a more elegant approach to guest checkout -> no email address
+    let email = shop.emails[0].address || "noreply@localhost";
+
+    if (buyer.emails.length > 0) {
+      if (buyer.emails[0].address) {
+        email = buyer.emails[0].address;
+      }
+    }
+
+    // format the address for Shippo
+    const shippoAddress = createShippoAddress(address, email, "QUOTE");
+    // make sure we mark this as an address validation for the Shippo API
+    shippoAddress.validate = true;
+
+    try {
+      const apiResult = ShippoApi.methods.validateAddress.call({ address: shippoAddress, apiKey });
+      if (apiResult.errors) {
+        return apiResult;
+      }
+      const validateAddress = createValidatedAddress(apiResult);
+      return validateAddress;
+    } catch (error) {
+      Logger.error(error);
+      throw new Meteor.Error(error, "Shippo API error");
```

Comment:
The form here is "short description" for first parameter and then the longer description for the second